robot/control/Inc/modules/LEDModule.hpp

```diff
@@ -130,9 +194,62 @@ class LEDModule : public GenericModule {
     LockedStruct<FPGAStatus>& fpgaStatus;
     LockedStruct<KickerInfo>& kickerInfo;
     LockedStruct<RadioError>& radioError;
+    LockedStruct<IMUData>& imuData;
 
     DigitalOut dotStarNCS;
 
+    // RADIO
+    const struct Error ERR_RADIO_BOOT_FAIL = {errorColors_0::RADIO_ERROR,
+                                              errorColors_1::FATAL,
+                                              errorColors_2::BOOT_FAIL};
+    const struct Error ERR_RADIO_WIFI_FAIL = {errorColors_0::RADIO_ERROR,
+                                              errorColors_1::FATAL,
+                                              errorColors_2::RADIO_CONN_WIFI_FAIL};
+    const struct Error ERR_RADIO_SOCCER_FAIL = {errorColors_0::RADIO_ERROR,
+                                                errorColors_1::FATAL,
+                                                errorColors_2::RADIO_CONN_SOCCER_FAIL};
+    // FPGA
+    const struct Error ERR_FPGA_BOOT_FAIL = {errorColors_0::FPGA_ERROR,
+                                             errorColors_1::FATAL,
+                                             errorColors_2::BOOT_FAIL};
+
+    // KICKER
+    const struct Error ERR_KICKER_BOOT_FAIL = {errorColors_0::KICKER_ERROR,
+                                               errorColors_1::FATAL,
+                                               errorColors_2::BOOT_FAIL};
+    // IMU
+    const struct Error ERR_IMU_BOOT_FAIL =  {errorColors_0::IMU_ERROR,
+                                             errorColors_1::FATAL,
+                                             errorColors_2::BOOT_FAIL};
+
+
+    /**
+     * Vector of colors for dotStars to display sequentially based on current errors
+     */
+    std::vector<Error> colorQueue;
```

Comment:
I wonder if there's a way to make this a static list with booleans/mask to turn on and off errors. This way you don't have to copy and remove from the vector continuously since each ERR should only exist once

Reply:
I think that what you gain in memory efficiency you lose in having to find the next element in the array which is true